aws/nat_gateway.go

```diff
@@ -0,0 +1,156 @@
+package aws
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/gruntwork-io/go-commons/retry"
+	multierror "github.com/hashicorp/go-multierror"
+
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+)
+
+func getAllNatGateways(session *session.Session, excludeAfter time.Time, configObj config.Config) ([]*string, error) {
+	svc := ec2.New(session)
+
+	allNatGateways := []*string{}
+	input := &ec2.DescribeNatGatewaysInput{}
+	err := svc.DescribeNatGatewaysPages(
+		input,
+		func(page *ec2.DescribeNatGatewaysOutput, lastPage bool) bool {
+			for _, ngw := range page.NatGateways {
+				if shouldIncludeNatGateway(ngw, excludeAfter, configObj) {
+					allNatGateways = append(allNatGateways, ngw.NatGatewayId)
+				}
+			}
+			return !lastPage
+		},
+	)
+	return allNatGateways, errors.WithStackTrace(err)
+}
+
+func shouldIncludeNatGateway(ngw *ec2.NatGateway, excludeAfter time.Time, configObj config.Config) bool {
+	if ngw == nil {
+		return false
+	}
+
+	if ngw.CreateTime != nil && excludeAfter.Before(*ngw.CreateTime) {
```

Comment:
What if `CreateTime` is `nil`? What does that mean?

Reply:
I think that is an impossible condition (since it's an autopopulated field), but I added this guard to avoid a panic. If it is `nil`, this routine will consider it in the nuke routine, which seems like the right thing to do given the impossibility of the condition.

Reply:
Roger